[Card Game 3] Просмотр стопок карт

SenX51 opened the pull request:
**Выполнил:**
 > Семенов А.В.

**Задача 2:**
 > Реализовать возможность просмотра стопок карт (колода, добор, сброс). При нажатии на карту в просмотре увеличить её. Закрытие по нажатию вне области.
___
**Демонстрация:**
### [**Видео**](https://github.com/user-attachments/assets/7465f196-d2c9-4df0-abe7-418ed2a6b018)

![image](https://github.com/user-attachments/assets/8b53f6cd-bc20-4630-b3a2-697b8f3c6032)
![image](https://github.com/user-attachments/assets/c48ed97f-3888-4bff-999d-ffc4f42c4847)
![image](https://github.com/user-attachments/assets/fe3a0650-0833-4cda-a4c4-bf79d06cce91)

SenX51 committed:
Card Library

 > Added ability to check card piles
 > Mocked-up battle page

SenX51 committed:
Styling/Naming fixes

Renamed showCardLibrary -> showCardPile
Deleted empty lines

SenX51 committed:
Added glow to hovered card

Added hover handling inside card pile
Added glow to hovered cards

SenX51 committed:
Prevented cards from being dragged or selected

SenX51 committed:
Disabled card right click

Disabled card right click, added emit to handle it instead.
Fixed card maintaining hover behaviour when enlrarging it in CardPile

reznikovkg commented on vue/src/components/card/CardPile.vue:
```diff
@@ -0,0 +1,160 @@
+      </div>
+      <div v-if="showingCard"
+        class="card-showcase center-horizontal center-vertical"
+        @click.self="() => showingCard = null"
```
Вынести в метод

reznikovkg commented on vue/src/components/card/CardPile.vue:
```diff
@@ -0,0 +1,160 @@
+        <CardComponent
+          :card="showingCard"
+          :style="enlargedCard"
+          />
```
Подвигать и выровнить

reznikovkg commented on vue/src/components/card/CardPile.vue:
```diff
@@ -0,0 +1,160 @@
+  methods: {
+    applyTransform(index) {
+      const isHovered = this.hoveredIndex == index;
+      if (isHovered)
```
Обязательно ставить { }

reznikovkg commented on vue/src/components/card/CardPile.vue:
```diff
@@ -0,0 +1,160 @@
+    background: #72727259
+  }
+
+  *::-webkit-scrollbar {
```
Убрать *

reznikovkg commented on vue/src/components/deck/DeckComponent.vue:
```diff
@@ -58,6 +58,7 @@ export default {
         return {
           transform: `scale(${this.getScaling + 0.05}) rotate(0deg) translateY(-35px)`,
           transition: `transform 0.1s ease-out`,
```
Почему не в классе?

SenX51 replied:
Находится в условной логике

reznikovkg commented on vue/src/components/pages/BattlePage.vue:
```diff
@@ -0,0 +1,114 @@
+
+    <div class="footer">
+      <div class="btn"
+        @click="() => showCardPile(getDrawPile)">
```
Поправить

reznikovkg commented on vue/src/components/pages/BattlePage.vue:
```diff
@@ -0,0 +1,114 @@
+
+<script>
+import { mapGetters } from "vuex";
+import CardLibrary from '../card/CardPile.vue';
```
Использовать названия файлов

reznikovkg commented on vue/src/components/pages/BattlePage.vue:
```diff
@@ -0,0 +1,114 @@
+  },
+  methods: {
+    showCardPile(cards) {
+      this.$refs.cardLibrary.setCardList(cards);
```
Переделать

SenX51 committed:
ref => bus, code styling

 > Changed parent->child event handling from ref to bus
 > Implemented setShowingCard() method
 > Fixed styling where neccessary

SenX51 committed:
Removed redundant code

SenX51 commented:
@reznikovkg добрый день, можете замержить PR, чтобы я мог создать новый и он не подтягивал изменения из этого

reznikovkg commented on vue/src/components/card/CardPile.vue:
```diff
@@ -0,0 +1,172 @@
+    };
+  },
+  mounted() {
+    this.bus.$on('setPile', (cards) => this.setCardList(cards));
```
this.$root.$on

reznikovkg replied:
this.$root.$emit('name', )

reznikovkg commented:
Сдано
